fe/fe-core/src/main/java/com/starrocks/scheduler/BaseTaskRunProcessor.java

```diff
@@ -0,0 +1,107 @@
+import com.starrocks.metric.ResourceGroupMetricMgr;
+import com.starrocks.plugin.AuditEvent;
+import com.starrocks.proto.PQueryStatistics;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.QueryState;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.service.FrontendOptions;
+import com.starrocks.sql.common.SqlDigestBuilder;
+import org.apache.commons.codec.binary.Hex;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public abstract class BaseTaskRunProcessor implements TaskRunProcessor {
+    @Override
+    public void processTaskRun(TaskRunContext context) throws Exception {
+        throw new NotImplementedException("Method processTaskRun need to implement");
+    }
+
+    protected void auditAfterExec(TaskRunContext context, StatementBase parsedStmt, PQueryStatistics statistics, boolean isMv) {
+        String origStmt = context.getDefinition();
+        ConnectContext ctx = context.getCtx();
+
+        // slow query
+        long endTime = System.currentTimeMillis();
+        long elapseMs = endTime - ctx.getStartTime();
+
+        ctx.getAuditEventBuilder().setEventType(AuditEvent.EventType.AFTER_QUERY)
+                .setState(ctx.getState().toString()).setErrorCode(ctx.getErrorCode()).setQueryTime(elapseMs)
+                .setScanBytes(statistics == null ? 0 : statistics.scanBytes)
+                .setScanRows(statistics == null ? 0 : statistics.scanRows)
+                .setCpuCostNs(statistics == null || statistics.cpuCostNs == null ? 0 : statistics.cpuCostNs)
+                .setMemCostBytes(statistics == null || statistics.memCostBytes == null ? 0 : statistics.memCostBytes)
+                .setReturnRows(ctx.getReturnRows())
+                .setStmtId(ctx.getStmtId())
+                .setQueryId(ctx.getQueryId() == null ? "NaN" : ctx.getQueryId().toString());
+
+        if (ctx.getState().isQuery()) {
+            MetricRepo.COUNTER_QUERY_ALL.increase(1L);
+            ResourceGroupMetricMgr.increaseQuery(ctx, 1L);
+            if (ctx.getState().getStateType() == QueryState.MysqlStateType.ERR) {
+                // err query
+                MetricRepo.COUNTER_QUERY_ERR.increase(1L);
+                ResourceGroupMetricMgr.increaseQueryErr(ctx, 1L);
+            } else {
+                // ok query
+                MetricRepo.COUNTER_QUERY_SUCCESS.increase(1L);
+                MetricRepo.HISTO_QUERY_LATENCY.update(elapseMs);
+                if (elapseMs > Config.qe_slow_log_ms || ctx.getSessionVariable().isEnableSQLDigest()) {
+                    MetricRepo.COUNTER_SLOW_QUERY.increase(1L);
+                    ctx.getAuditEventBuilder().setDigest(computeStatementDigest(parsedStmt));
+                }
+            }
+            ctx.getAuditEventBuilder().setIsQuery(true);
+        } else {
+            ctx.getAuditEventBuilder().setIsQuery(false);
+        }
+        ctx.getAuditEventBuilder().setIsMv(isMv);
```

Comment:
isMV may not better mark. I think better to use isSystem.
By the way. the refresh of the materialized view may not need to record audit.log, but CTAS needs to be recorded, because the materialized view is refreshed at the partition level, if it is recorded, there may be a lot of logs.

Reply:
could we add the type field here?  type could include: query, insert, MV, statistic and so on